crates/consensus/src/transaction/typed.rs

```diff
@@ -78,7 +78,7 @@ impl From<TxEnvelope> for TypedTransaction {
 
 impl TypedTransaction {
     /// Return the [`TxType`] of the inner txn.
-    pub const fn tx_type(&self) -> TxType {
+    pub const fn transaction_type(&self) -> TxType {
```

Comment:
should the type `TxType` also be renamed?

Reply:
I tend to agree, it is however one of the rabbit holes I fell down when drafting the PR

There are quite a few structs following that naming pattern that I suggest we change:

- `TxEip1559` -> `Eip1559Transaction`
- `TxLegacy` -> `LegacyTransaction`
- `TxEnvelope` -> `TransactionEnvelope`
- `TxType` -> `TransactionType`
- `AnyTxType` -> `AnyTransactionType`
- `TxReceipt` -> `TransactionReceipt`
- `TxSigner` -> `TransactionSigner`
- `TxSignerSync` -> `TransactionSignerSync`
- `SendableTx` -> `SendableTransaction`
- `TxFiller` -> `TransactionFiller`
- `TxGasAndReward` -> `TransactionGasAndReward`
- `UnsignedTx` -> `UnsignedTransaction`
- `Signed` -> `SignedTransaction`

In line with the proposed changes in the PR I would suggest we change in `core`:

- `TxKind` -> `TransactionKind`
- `TxHash` -> `TransactionHash`
- `TxIndex` -> TransactionIndex`
- `TxNumber` -> `TransactionNumber`

Reply:
i don't love making every type 9 characters longer 😮‍💨

Reply:
I think so.

Reply:
If everything is already on `Tx` instead of `Transaction` why are we changing it to the more verbose version?

Reply:
meh, so maybe not rename anything.
don't love `tx_` on functions but it's better than renaming everything